src/api/nodeRpcCall.js

```diff
@@ -913,52 +1012,60 @@ const getOfficialUserRegistryEntries = async (walletAddress) => {
   let registeredCompaniesByWallet = []
   let supportedObject = JSON.parse(JSON.stringify(newCompanyDataObject))
   companyRegistryRawData.forEach((companyRegistryEntity, index) => {
-    const dataToHuman = companyRegistryEntity.toHuman()
-    if(dataToHuman){
-      let companyData = JSON.parse(dataToHuman.data)
-      const staticFields = []
-      const dynamicFields = []
-      dataToHuman.data = companyData
-      supportedObject.staticFields.forEach(staticField => {
-        if (staticField.key in companyData){
-          let fieldObject = staticField
-          fieldObject.display = companyData[staticField.key]
-          staticFields.push(fieldObject)
-        }
-      })
-      supportedObject.dynamicFields.forEach(dynamicField => {
-        if(dynamicField.key in companyData){
-          let fieldObject = dynamicField
-          let fieldObjectData = []
-          companyData[dynamicField.key].forEach(dynamicFieldDataArray => {
-            //Format using fields data
-            let crossReferencedFieldDataArray = []
-            for(const key in dynamicFieldDataArray){
-              if(!key.includes('IsEncrypted')){
-                let pushObject = {}
-                let keyIsEncrypted = key + 'IsEncrypted'
-                pushObject['key'] = key
-                pushObject['display'] = dynamicFieldDataArray[key]
-                pushObject['isEncrypted'] = dynamicFieldDataArray[keyIsEncrypted]
-                crossReferencedFieldDataArray.push(JSON.parse(JSON.stringify(pushObject)))
-              }
+    let companyData;
+    try {
+      const compressed = companyRegistryEntity.unwrap().data;
+      companyData = api.createType('CompanyData', pako.inflate(compressed)).toJSON();
+    } catch(e) {
+      console.error("Invalid company data", e); // FIXME actually display this to user somehow?
+      return;
```

Comment:
Shouldn't we return null?

Reply:
Not sure it matters.

1. It should never happen
2. Both `undefined` (which I think will result in infinite loader) and `null` (which I think will result in "No registered companies") are wrong. I kinda thing `null` is worse, as it just lies to user, while `undefined` at least will clearly show as something that's broken. We just need a proper error handling for this.

I've created BLOCKCHAIN-254 ticket to implement the error handling. I don't want to modify this file, as there are other PRs that already rely on it, so I want to limit changes to critical only.